src/tests/lvm/lvm_ha_controller/lvm_ha_controller_test.go

```diff
@@ -0,0 +1,274 @@
+	Expect(err).To(BeNil(), "failed to scale deployment %s, error: %v", lvmControllerName, err)
+
+	//verify pvc and pv to be bound
+	volUuid, err := k8stest.VerifyVolumeProvision(app.GetPvcName(), common.NSDefault)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(volUuid).ToNot(BeEmpty())
+
+	// use created PVC which is deployed as part of restore app
+	err = app.RefreshVolumeState()
+	Expect(err).ToNot(HaveOccurred())
+
+	// deploy fio pod with created volume
+	logf.Log.Info("deploy fio pod with created volume")
+	err = app.DeployApplication()
+	Expect(err).To(BeNil(), "failed to deploy app")
+
+	// Get the name of the controller pod replica which is active as master at present
+	lease, err := k8stest.GetLease(e2e_config.LvmEngineLeaseName, common.NSOpenEBS())
+	Expect(err).To(BeNil(), "failed to get lease %s in kube-system namespace", e2e_config.LvmEngineLeaseName)
+	Expect(lease).ToNot(BeNil(), "no lease found")
+
+	// get lvm controller master pod i.e spec.holderIdentity from lease
+	initialHolderIdentity := lease.Spec.HolderIdentity
+	Expect(initialHolderIdentity).ToNot(BeNil(), "no lease HolderIdentity found")
+	logf.Log.Info("Lvm controller", "Initial HolderIdentity", initialHolderIdentity)
+
+	// taint all nodes so that after deleting one of the lvm controller pods, now pod should not scheduled
+	for _, node := range nodesWithoutTaint {
+		err = k8stest.AddNoScheduleTaintOnNode(node)
+		Expect(err).To(BeNil(), "failed to taint node %s", node)
+	}
+
+	// delete lvm controller pod which is holding lease
+	err = k8stest.DeletePod(*initialHolderIdentity, common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "fio to delete pod %s", *initialHolderIdentity)
+
+	// wait for lease to switch to different pod
+	Eventually(func() bool {
+		lease, err := k8stest.GetLease(e2e_config.LvmEngineLeaseName, common.NSOpenEBS())
+		if err != nil {
+			logf.Log.Info("failed to get lease in kube-system namespace", "lease name", e2e_config.LvmEngineLeaseName, "error", err)
+			return false
+		}
+		if *lease.Spec.HolderIdentity == "" {
+			logf.Log.Info("lease HolderIdentity can not be empty string")
+			return false
+		} else if *lease.Spec.HolderIdentity != *initialHolderIdentity {
+			return true
+		}
+
+		return false
+	},
+		defFioCompletionTime,
+		"5s",
+	).Should(BeTrue())
+	Expect(err).To(BeNil(), "failed to get lease")
+
+	// Check fio pod status
+	phase, podLogSynopsis, err := k8stest.CheckFioPodCompleted(app.GetPodName(), common.NSDefault)
+	Expect(err).To(BeNil(), "CheckPodComplete got error %s", err)
+	Expect(phase).ShouldNot(Equal(coreV1.PodFailed), "fio pod phase is %s, %s", phase, podLogSynopsis)
+
+	// remove app pod, pvc,sc
+	err = app.Cleanup()
+	Expect(err).To(BeNil(), "failed to clean resources")
+
+	// remove taints form nodes
+	for _, node := range nodesWithoutTaint {
+		err = k8stest.RemoveNoScheduleTaintFromNode(node)
+		Expect(err).To(BeNil(), "failed to taint node %s", node)
+	}
+	nodesWithoutTaint = []string{}
```

Comment:
may be after removing taint we can verify that pending controller pod comes into running state and total replicas are now original+1